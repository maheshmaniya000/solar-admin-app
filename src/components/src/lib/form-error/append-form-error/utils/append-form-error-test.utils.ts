import { ElementRef, Type } from '@angular/core';
import {
  AbstractControl,
  AsyncValidatorFn,
  ValidationErrors
} from '@angular/forms';
import { SpectatorDirective } from '@ngneat/spectator/jest';
import { isEmpty } from 'lodash-es';
import { Observable, of } from 'rxjs';
import { delay } from 'rxjs/operators';

import { AppendFormErrorDirective } from '../append-form-error.directive';

export class AppendFormErrorTestUtils {
  static readonly asyncRequired: (
    delayInMilliseconds: number
  ) => AsyncValidatorFn =
    (delayInMilliseconds: number) =>
    (control: AbstractControl): Observable<ValidationErrors> =>
      isEmpty(control.value)
        ? of({ required: true }).pipe(delay(delayInMilliseconds))
        : of(null);

  static expectDirectiveToBePresentOn<D extends AppendFormErrorDirective>(
    element: Element,
    type: Type<D>,
    spectator: SpectatorDirective<AppendFormErrorDirective>
  ): void {
    const appendFormErrorDirectiveElement: Element = spectator.query(type, {
      read: ElementRef
    }).nativeElement;
    expect(appendFormErrorDirectiveElement).toBe(element);
  }
}
