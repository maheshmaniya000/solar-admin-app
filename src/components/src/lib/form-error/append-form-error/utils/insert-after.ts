export function insertAfter(newChild: Element, refChild: Element): void {
  refChild.parentNode.insertBefore(newChild, refChild.nextSibling);
}
