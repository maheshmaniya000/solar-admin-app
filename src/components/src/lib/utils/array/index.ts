export * from './array-min-max';
export * from './ensure-index-in-array-bounds';
