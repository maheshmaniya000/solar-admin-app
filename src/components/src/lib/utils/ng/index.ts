export * from './extract-touched-changes';
