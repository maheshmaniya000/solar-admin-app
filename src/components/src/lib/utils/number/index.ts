export * from './angle-in-range-to-label';
export * from './decimal-places';
export * from './ranges';
export * from './factorize';
export * from './is-close-to';
export * from './multiple-of';
export * from './is-natural-number';
export * from './is-natural-number-or-zero';
export * from './split-to-similar-values';
export * from './rounding';
export * from './units';
