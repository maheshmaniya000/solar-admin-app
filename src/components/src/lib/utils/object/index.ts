export * from './is-not-empty';
export * from './is-not-nil';
