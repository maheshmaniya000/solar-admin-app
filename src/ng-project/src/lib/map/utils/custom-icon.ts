import * as L from 'leaflet';

type CustomIconType = ((oldIcon: any) => void) & L.Icon;

export const customIcon = L.Icon.extend({
  options: {
    nativeElement: null,
  },

  createIcon() {
    const element = this.options.nativeElement;
    const div = document.createElement('div');
    // eslint-disable-next-line no-underscore-dangle
    this._setIconStyles(div, 'icon');
    div.appendChild(element);

    return div;
  },

  createShadow() {
    return null;
  }
}) as any as CustomIconType;
