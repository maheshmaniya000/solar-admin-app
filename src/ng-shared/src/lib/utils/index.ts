export * from './router/custom.reuse-stragegy';
export * from './router/custom.router-state.serializer';
export * from './router/custom.url-serializer';
export * from './rxjs/simplechange.operator';
export * from './app-device.utils';
export * from './dom.utils';
export * from './enum.utils';
export * from './misc';
export * from './module.utils';
export * from './response';
export * from './timezone.utils';
export * from './translation.utils';
export * from './validators';
