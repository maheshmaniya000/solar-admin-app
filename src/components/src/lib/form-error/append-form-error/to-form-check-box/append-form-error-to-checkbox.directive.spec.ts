import { ElementRef } from '@angular/core';
import {
  FormControl,
  FormsModule,
  ReactiveFormsModule,
  Validators
} from '@angular/forms';
import { MatCheckbox, MatCheckboxModule } from '@angular/material/checkbox';
import {
  createDirectiveFactory,
  SpectatorDirective
} from '@ngneat/spectator/jest';
import { TranslocoTestingModule } from '@ngneat/transloco';

import { AppendFormErrorTestUtils } from '../utils/append-form-error-test.utils';
import { AppendFormErrorToCheckboxDirective } from './append-form-error-to-checkbox.directive';

describe('AppendFormErrorToCheckboxDirective', () => {
  describe('component', () => {
    const createDirective = createDirectiveFactory({
      directive: AppendFormErrorToCheckboxDirective,
      imports: [
        ReactiveFormsModule,
        FormsModule,
        MatCheckboxModule,
        TranslocoTestingModule.forRoot({ langs: {} })
      ],
      shallow: true
    });
    let spectator: SpectatorDirective<AppendFormErrorToCheckboxDirective>;

    describe('directive presence', () => {
      it(`should not attach directive to mat-checkbox element without
 sgAppendFormError `, () => {
        expect(() =>
          createDirective(`
            <mat-checkbox ngModel></mat-checkbox>
          `)
        ).toThrow();
      });

      it(`should attach directive to mat-checkbox element without
 sgAppendFormError `, () => {
        spectator = createDirective(`
          <mat-checkbox ngModel sgAppendFormError></mat-checkbox>
        `);
        AppendFormErrorTestUtils.expectDirectiveToBePresentOn(
          spectator.query(MatCheckbox, { read: ElementRef }).nativeElement,
          AppendFormErrorToCheckboxDirective,
          spectator
        );
      });
    });

    describe('error message rendering', () => {
      it(`should render error message under invalid and touched checkbox`, () => {
        const inputElementName = 'input';
        const formControl = new FormControl(undefined, [Validators.required]);
        spectator = createDirective(
          `
            <mat-checkbox
              [formControl]="formControl"
              sgAppendFormError
            >
            </mat-checkbox>`,
          {
            hostProps: { formControl }
          }
        );
        spectator.focus(inputElementName);
        spectator.blur(inputElementName);

        expect(spectator.fixture).toMatchSnapshot();
      });
    });
  });
});
