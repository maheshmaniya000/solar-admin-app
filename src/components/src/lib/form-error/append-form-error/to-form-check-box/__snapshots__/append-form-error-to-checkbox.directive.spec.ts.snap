// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppendFormErrorToCheckboxDirective component error message rendering should render error message under invalid and touched checkbox 1`] = `
<ng-component
  __ngContext__={[Function LRootView]}
  formControl={[Function FormControl]}
>
  <mat-checkbox
    class="mat-checkbox mat-accent _mat-animation-noopable ng-pristine ng-invalid ng-touched"
    sgappendformerror=""
  >
    <label
      class="mat-checkbox-layout"
      for="mat-checkbox-3-input"
    >
      <span
        class="mat-checkbox-inner-container mat-checkbox-inner-container-no-side-margin"
      >
        <input
          aria-checked="false"
          class="mat-checkbox-input cdk-visually-hidden"
          id="mat-checkbox-3-input"
          tabindex="0"
          type="checkbox"
        />
        <span
          class="mat-ripple mat-checkbox-ripple mat-focus-indicator"
          matripple=""
        >
          <span
            class="mat-ripple-element mat-checkbox-persistent-ripple"
          />
        </span>
        <span
          class="mat-checkbox-frame"
        />
        <span
          class="mat-checkbox-background"
        >
          <svg
            aria-hidden="true"
            class="mat-checkbox-checkmark"
            focusable="false"
            version="1.1"
            viewBox="0 0 24 24"
            xml:space="preserve"
          >
            <path
              class="mat-checkbox-checkmark-path"
              d="M4.1,12.7 9,17.6 20.3,6.3"
              fill="none"
              stroke="white"
            />
          </svg>
          <span
            class="mat-checkbox-mixedmark"
          />
        </span>
      </span>
      <span
        class="mat-checkbox-label"
      >
        <span
          style="display: none;"
        >
        </span>
      </span>
    </label>
  </mat-checkbox><sg-form-error>
    <mat-error>
       en.common.validation.required 
    </mat-error>
  </sg-form-error>
</ng-component>
`;
